presto-main/src/main/java/com/facebook/presto/operator/NestedLoopJoinOperator.java

```diff
@@ -193,18 +201,37 @@ public Page getOutput()
         }
 
         if (nestedLoopPageBuilder != null && nestedLoopPageBuilder.hasNext()) {
-            return nestedLoopPageBuilder.next();
+            return applyFilter(nestedLoopPageBuilder.next());
         }
 
         if (buildPageIterator.hasNext()) {
             nestedLoopPageBuilder = new NestedLoopPageBuilder(probePage, buildPageIterator.next());
-            return nestedLoopPageBuilder.next();
+            return applyFilter(nestedLoopPageBuilder.next());
         }
 
         probePage = null;
         return null;
     }
 
+    private Page applyFilter(Page page)
+    {
+        if (!pageFilter.isPresent()) {
+            return page;
+        }
+
+        SelectedPositions selectedPositions = pageFilter.get().filter(operatorContext.getSession().toConnectorSession(), pageFilter.get().getInputChannels().getInputChannels(page));
```

Comment:
```java
if (selectedPositions.isEmpty()) {
    return null;
}
```

Or consider making `applyFilter` return `Optional<Page>`... This method is only called from `getOutput` so returning null is ok.  I'm not sure which form is easier to read.